### PR TITLE
Absolute links across reference guides

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [ ] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [ ] Unit or Functional tests are included in the PR

**Description:**

For now at least make any reference guide links absolute for the target they will be when deployed to the site.

In the future we perhaps could be able to do this automagically.

This doesn't cover all the links I'd guess but hopefully a start.

### CODE REVIEW

File: docs/en/creating-widgets/supplemental.md [hunk 1]
```diff
@@ -155,11 +155,11 @@ Dojo recognizes two types of nodes within its VDOM:
...
+Both `VNode`s and `WNode`s are considered subtypes of `DNode`s within Dojo's virtual DOM, but applications don't typically deal with `DNode`s in their abstract sense. Using [TSX syntax](/learn/creating-widgets/rendering-widgets#tsx-support) is also preferred as it allows applications to render both virtual node types with uniform syntax.
 
 ### Instantiating VDOM nodes
 
-If [TSX output](#tsx-support) is not desired, widgets can import one or both of the `v()` and `w()` primitives provided by the `@dojo/framework/core/vdom` module. These create `VNode`s and `WNode`s, respectively, and can be used as part of the return value from a [widget's render function](#basic-widget-structure). Their signatures, in abstract terms, are:
+If TSX output is not desired, widgets can import one or both of the `v()` and `w()` primitives provided by the `@dojo/framework/core/vdom` module. These create `VNode`s and `WNode`s, respectively, and can be used as part of the return value from a [widget's render function]./learn/creating-widgets/widget-fundamentals#basic-widget-structure). Their signatures, in abstract terms, are:
```
Comment (sbinge) [inline docs/en/creating-widgets/supplemental.md]: Missing (

Comment (sbinge) [inline docs/en/creating-widgets/supplemental.md]:
```suggestion
If TSX output is not desired, widgets can import one or both of the `v()` and `w()` primitives provided by the `@dojo/framework/core/vdom` module. These create `VNode`s and `WNode`s, respectively, and can be used as part of the return value from a [widget's render function](./learn/creating-widgets/widget-fundamentals#basic-widget-structure). Their signatures, in abstract terms, are:
```

File: docs/en/creating-widgets/supplemental.md [hunk 2]
```diff
@@ -523,15 +523,15 @@ export default class FocusableWidget extends Focus(WidgetBase) {
 
 # Managing state
 
-For simple applications where data is not required to flow between many components, state management can be very straightforward. Data can be encapsulated within individual widgets that need it as the most [basic form of state management](#basic-internal-widget-state) within a Dojo application.
+For simple applications where data is not required to flow between many components, state management can be very straightforward. Data can be encapsulated within individual widgets that need it as the most [basic form of state management](/learn/creating-widgets/managing-state#basic-internal-widget-state) within a Dojo application.
```
Comment (sbinge) [inline docs/en/creating-widgets/supplemental.md]:
```suggestion
For simple applications where data is not required to flow between many components, state management can be very straightforward. Data can be encapsulated within individual widgets that need it as the most [basic form of state management](/learn/creating-widgets/managing-state#basic-self-encapsulated-widget-state) within a Dojo application.
```

File: docs/en/creating-widgets/supplemental.md [hunk 3]
```diff
@@ -523,15 +523,15 @@ export default class FocusableWidget extends Focus(WidgetBase) {
...
 
-As applications grow in complexity and start requiring data to be shared and transferred between multiple widgets, a more robust form of state management is required. Here, Dojo begins to prove its value as a reactive framework, allowing applications to define how data should flow between components, then letting the framework manage change detection and re-rendering. This is done by [wiring widgets and properties together](#intermediate-widget-properties) when declaring VDOM output in a widget's render function.
+As applications grow in complexity and start requiring data to be shared and transferred between multiple widgets, a more robust form of state management is required. Here, Dojo begins to prove its value as a reactive framework, allowing applications to define how data should flow between components, then letting the framework manage change detection and re-rendering. This is done by [wiring widgets and properties together]./learn/creating-widgets/managing-state#intermediate-widget-properties) when declaring VDOM output in a widget's render function.
```
Comment (sbinge) [inline docs/en/creating-widgets/supplemental.md]:
```suggestion
As applications grow in complexity and start requiring data to be shared and transferred between multiple widgets, a more robust form of state management is required. Here, Dojo begins to prove its value as a reactive framework, allowing applications to define how data should flow between components, then letting the framework manage change detection and re-rendering. This is done by [wiring widgets and properties together]./learn/creating-widgets/managing-state#intermediate-passing-widget-properties) when declaring VDOM output in a widget's render function.
```

File: docs/en/middleware/supplemental.md [hunk 4]
```diff
@@ -205,7 +205,7 @@ import icache from '@dojo/framework/core/middleware/icache';
 
 Allows widgets to theme their CSS classes when rendering, and also provides applications the ability to set themes and determine what the currently set theme is, if any.
 
-Described in detail in the [Dojo Styling and Theming reference guide](../styling-and-theming/supplemental.md#making-themeable-widgets).
+Described in detail in the [Dojo Styling and Theming reference guide](/learn/theming/theming-a-dojo-application#making-themeable-widgets).
```
Comment (sbinge) [inline docs/en/middleware/supplemental.md]:
```suggestion
Described in detail in the [Dojo Styling and Theming reference guide](/learn/styling/theming-a-dojo-application#making-themeable-widgets).
```

File: docs/en/middleware/supplemental.md [hunk 5]
```diff
@@ -377,7 +377,7 @@ import store from '@dojo/framework/core/middleware/store';
 
 ## `focus`
 
-Allows widgets to inspect and control focus amongst their resulting DOM output when combined with the [VDOM focus primitives](../creating-widgets/supplemental.md#handling-focus).
+Allows widgets to inspect and control focus amongst their resulting DOM output when combined with the [VDOM focus primitives](/learn/creating-widgets/interactivity#handling-focus).
```
Comment (sbinge) [inline docs/en/middleware/supplemental.md]:
```suggestion
Allows widgets to inspect and control focus amongst their resulting DOM output when combined with the [VDOM focus primitives](/learn/creating-widgets/enabling-interactivity#handling-focus).
```

File: docs/en/middleware/supplemental.md [hunk 6]
```diff
@@ -386,7 +386,7 @@ import focus from '@dojo/framework/core/middleware/focus';
 ```
 
 -   `focus.shouldFocus(): boolean`
-    -   Returns `true` if focus should be specified within the current render cycle. Will only return `true` once, after which `false` is returned from future calls until `focus.focus()` is called again. This function is typically passed as the [`focus` property](../creating-widgets/supplemental.md#handling-focus) to a specific VDOM node, allowing the widget to direct where focus should be applied.
+    -   Returns `true` if focus should be specified within the current render cycle. Will only return `true` once, after which `false` is returned from future calls until `focus.focus()` is called again. This function is typically passed as the focus` property to a specific VDOM node, allowing the widget to direct where focus should be applied.
```
Comment (sbinge) [inline docs/en/middleware/supplemental.md]:
```suggestion
    -   Returns `true` if focus should be specified within the current render cycle. Will only return `true` once, after which `false` is returned from future calls until `focus.focus()` is called again. This function is typically passed as the `focus` property to a specific VDOM node, allowing the widget to direct where focus should be applied.
```

File: docs/en/middleware/supplemental.md [hunk 7]
```diff
@@ -489,15 +489,15 @@ import injector from '@dojo/framework/core/middleware/injector';
...
+    -   Subscribes the given `callback` invalidation function against the specified registry `label` injector (if one exists). If a `callback` is not specified, the [`invalidator`](/learn/middleware/core-render-middleware#invalidator) middleware is used by default so that the current widget will be marked as invalid and re-rendered when the injector makes its data available.
 -   `injector.get<T>(label: RegistryLabel): T | null`
     -   Retrieves the current injector associated with the given registry `label`, or `null` if no such injector exists.
 
 ## `block`
 
 Allows widgets to execute modules known as **blocks** within Node.js at build time. Typically used as part of build-time rendering.
 
-Described in detail in the [Building reference guide](../building/supplemental.md#build-time-rendering).
+Described in detail in the [Building reference guide](/learn/building/buildtime-rendering).
```
Comment (sbinge) [inline docs/en/middleware/supplemental.md]: Is this correct given how site is now generating URLs from headers? or should it be `build-time-rendering`?

File: docs/en/middleware/supplemental.md [hunk 8]
```diff
@@ -510,7 +510,7 @@ import block from '@dojo/framework/core/middleware/block';
 
 # Core render middleware
 
-The `@dojo/framework/core/vdom` module includes foundational middleware that is useful across the majority of Dojo applications. These are mainly useful when building other custom middleware (they underpin the [additional middleware](#optional-middleware) offered by the framework), but can occasionally be useful in general widget development.
+The `@dojo/framework/core/vdom` module includes foundational middleware that is useful across the majority of Dojo applications. These are mainly useful when building other custom middleware (they underpin the [additional middleware](/learn/middleware/available-middleware#optional-middleware) offered by the framework), but can occasionally be useful in general widget development.
```
Comment (sbinge) [inline docs/en/middleware/supplemental.md]:
```suggestion
The `@dojo/framework/core/vdom` module includes foundational middleware that is useful across the majority of Dojo applications. These are mainly useful when building other custom middleware (they underpin the [additional middleware](/learn/middleware/available-middleware) offered by the framework), but can occasionally be useful in general widget development.
```

File: docs/en/middleware/supplemental.md (hunk 7 above)
Comment (agubler) [reply docs/en/middleware/supplemental.md]: @sbinge yup it's correct, it renders as `buildtime-rendering`

File: docs/en/creating-widgets/introduction.md [hunk 9]
```diff
@@ -16,8 +16,8 @@ Widgets describe their intended structural representation through virtual nodes
 
 ## Defining a widget
 
--   Using the [`create()` primitive](./supplemental.md#basic-widget-structure) to define a widget as a render function factory
--   Returning [virtual DOM nodes](./supplemental.md#working-with-the-vdom) that define the widget's structural representation, declared as [TSX syntax](./supplemental.md#tsx-support)
+-   Using the [`create()` primitive](/learn/creating-widgets/widget-fundamentals#basic-widget-structure) to define a widget as a render function factory
+-   Returning [virtual DOM nodes](/learn/creating-widgets/rendering-widgets/#working-with-the-vdom) that define the widget's structural representation, declared as [TSX syntax](/learn/creating-widgets/rendering-widgets/#working-with-the-vdom)
```
Comment (sbinge) [inline docs/en/creating-widgets/introduction.md]:
```suggestion
-   Returning [virtual DOM nodes](/learn/creating-widgets/rendering-widgets/#working-with-the-vdom) that define the widget's structural representation, declared as [TSX syntax](/learn/creating-widgets/rendering-widgets#tsx-support)
```

**Pull request merged**